plugin/src/main/java/org/opensearch/ml/sdkclient/SdkClientModule.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.ml.sdkclient;
+
+import org.opensearch.OpenSearchException;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.client.transport.aws.AwsSdk2Transport;
+import org.opensearch.client.transport.aws.AwsSdk2TransportOptions;
+import org.opensearch.common.inject.AbstractModule;
+import org.opensearch.core.common.Strings;
+import org.opensearch.sdk.SdkClient;
+
+import software.amazon.awssdk.http.SdkHttpClient;
+import software.amazon.awssdk.http.apache.ApacheHttpClient;
+import software.amazon.awssdk.regions.Region;
+
+/**
+ * A module for binding this plugin's desired implementation of {@link SdkClient}.
+ */
+public class SdkClientModule extends AbstractModule {
+
+    public static final String REMOTE_METADATA_ENDPOINT = "REMOTE_METADATA_ENDPOINT";
+    public static final String REGION = "REGION";
+
+    private final String remoteMetadataEndpoint;
+    private final String region;
+
+    /**
+     * Instantiate this module using environment variables
+     */
+    public SdkClientModule() {
+        this(System.getenv(REMOTE_METADATA_ENDPOINT), System.getenv(REGION));
```

Comment:
Optionally can cluster settings be used to fetch metadata endpoint?

Reply:
Assuming you have easy access to the cluster settings from the plugin, yes.  In the current plugin environment we have settings available, but in earlier Extensions SDK work we had to create some overhead to subscribe to them.

Lots of ways to do this, I just based this implementation on [Your POC here](https://github.com/arjunkumargiri/ml-commons/commit/49e8f1ad95922f61e6be8b806a1808179f0c6c79#diff-c548f309dba4a5c4be8a5d19a7c7bd23649f7183c7a6819d585e779c49829d31).